src/dupin.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+#include <Eigen/Dense>
+
+
+// DynamicProgramming class for dynamic programming based segmentation.
+class DynamicProgramming { 

```

Comment:
Methods which perform some computation (other than getters and setters) should be action verbs not nouns this makes them easier to understand at a glance.